public/router.php

```diff
@@ -25,15 +25,21 @@
 	date_default_timezone_set('UTC');
 }
 
-define('BASE_DIR', dirname(__DIR__).'/');
+$baseDir = str_replace('\\', '/', dirname(dirname(__FILE__)).'/');
+if (!is_file($baseDir.'pufferpanel')) {
+    $baseUrl = dirname($_SERVER['PHP_SELF']);
```

Comment:
We already ship configs that do just that. This solution is not drop and go if you have to change folder structures to get it to work. I do not get why that is needed.

I *prefer* just changing configs. If it fails, you restore a config and it works. Moving folders around *WILL BREAK* when we do updates. This solution does not fully solve the issue. 

We need a way to solve this using just the nginx/apache configs and the master_url. Using any form of "folder layout" will be a nightmare for supporting setups.